agent/grpc-external/services/peerstream/stream_tracker.go

```diff
@@ -153,8 +167,11 @@ type Status struct {
 	// LastSendErrorMessage tracks the last error message when sending into the stream.
 	LastSendErrorMessage string
 
-	// LastReceiveSuccess tracks the time we last successfully stored a resource replicated FROM the peer.
-	LastReceiveSuccess time.Time
+	// LastReceiveHeartbeat tracks when we last received a heartbeat from our peer.
+	LastReceiveHeartbeat time.Time
+
+	// LastReceiveResourceSuccess tracks the time we last successfully stored a resource replicated FROM the peer.
+	LastReceiveResourceSuccess time.Time
```

Comment:
Were these renamed to include `Resource` due to separating Heartbeat vs Resource?

We may want to align `LastReceiveError` and `LastReceiveErrorMessage` to be qualified with `Resource` as well. And I personally wouldn't mind `Receive` shortened to `Recv` since the names are getting long 😆

Reply:
LastReceiveError and LastReceiveErrorMessage are still generic for both resources and heartbeats as well as other errors such as the stream being disconnected.

I can do the shortening for sure.

Reply:
I'll merge this as is, got a PR for the shortening and then I can come back to separating out the lastreceiveerror etc into LastReceiveError (for generic receive errors) and then LastReceiveResourceError (for replicated resource receive errors). Could also wait on that change until we figure out how we're going to surface these errors.